docs/requirements/ADP-957.md

```diff
@@ -0,0 +1,201 @@
+# Basic definitions
+
+*   **Collateral inputs**, in a transaction, are used to cover transaction fees
+    in the case that a non-native script fails. The term **collateral** refers
+    to the total ada contained in the UTxOs referenced by collateral inputs.
+    Collateral inputs can **only** contain ada: they may **not** contain
+    non-ada assets.
+
+*   The protocol parameter `collateralPercent` is used to specify the
+    percentage of the total transaction fee its collateral must (at a minimum)
+    cover. The collateral inputs must not themselves be locked by a script.
+
+*   The protocol parameter `maxCollateralInputs` is used to limit,
+    additionally, the total number of collateral inputs, and thus the total
+    number of additional signatures that must be checked during validation.
+
+# Requirements
+
+1.  The wallet application, when requested to create a transaction requiring
+    collateral, should attempt to select one or more **pure-ada** entries from
+    the wallet's UTxO set to serve as collateral inputs. The wallet application
+    **must not** select UTxO entries that include assets other than ada to
+    serve as collateral.
+
+2.  The wallet application, when requested to create a transaction requiring
+    collateral, should attempt to select no fewer than one UTxO entry, and no
+    more than `maxCollateralInputs` UTxO entries to serve as collateral.
+
+    If it is not possible to satisfy these bounds, the wallet application must
+    return an error to the caller.
+
+3.  The wallet application, when requested to create a transaction requiring
+    collateral, should attempt to select an amount of ada that is greater than
+    or equal to a minimum bound that is determined by the `collateralPercent`
+    protocol parameter and the transaction fee:
+
+        collateralAmount ≥ txFee × collateralPercent
+
+    If it is not possible to satisfy the minimum bound, the wallet application
+    must return an error to the caller.
+
+4.  The wallet application, when requested to create a transaction requiring
+    collateral, should select an amount of ada that is less than or equal to a
+    maximum bound that is determined by a user-specified `maxCollateralFactor`
```

Comment:
This isn't a requirement - it's a solution for a requirement.
The requirement is that we shouldn't put too much of the user's ada "at risk" of being lost by an invalid transaction.

Reply:
> The requirement is that we shouldn't put too much of the user's ada "at risk" of being lost by an invalid transaction.

Sure. Though my motivation for writing this document is to clarify exactly what terms like "too much" mean.

If we only say that the wallet "shouldn't put too much ada at risk", then how much is too much? It's not clear how we would measure whether the requirement has been satisfied.

My goal here has been to apply the [SMART principles](https://erivanramos.medium.com/there-is-a-smart-way-to-write-software-requirements-48add56d3972) where it makes sense. In particular, to make each requirement specific and measurable.